app/controllers/users_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+  def index
+    users = User.all
+
+    render json: users, status: 200
+  end
+
+  def show
+    user = User.find(params[:id])
+
+    render json: user, status: 200
+  end
+
+  def create
+    user = User.new(user_params)
+
+    if user.save!
+      render json: user, status: 201
+    else
+      render json: {error: "User is invalid"}, status: 400
+    end
+  end
+
+  def update
+    user = User.find(params[:id])
+
+    if user.update_attributes(user_params)
+      render json: user, status: 200
+    else
+      render json: {error: "User update failed", status: 400}, status: 400
```

Comment:
I am just updating the attributes so status 200 fits because it shows that the request was processed successfully.